File: examples/example-kubernetes-client-openapi-reactor-python/src/test/python/micronaut/client/PodControllerTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import ApplicationContext
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from micronaut.client.PodController import PodController


@MicronautTest
class PodControllerTest:
    context: Annotated[ApplicationContext, Inject] = None

    @Test
    def test_core_v1_api_reactor_is_injected(self):
        controller = self.context.getBean(PodController).asPolyglotValue()
        assert controller.core_v1_api_reactor is not None

File: examples/example-kubernetes-client-openapi-reactor-kotlin/src/test/kotlin/micronaut/client/PodControllerTest.kt
package micronaut.client

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test

@MicronautTest
class PodControllerTest {

    @Inject
    lateinit var controller: PodController

    @Test
    fun testCoreV1ApiReactorIsInjected() {
        assertNotNull(controller.coreV1ApiReactor)
    }
}
